common/testing/historyrequire/history_require.go

```diff
@@ -48,60 +48,70 @@ type (
 )
 
 var publicRgx = regexp.MustCompile("^[A-Z]")
+var typeOfBytes = reflect.TypeOf([]byte(nil))
 
 func New(t require.TestingT) HistoryRequire {
 	return HistoryRequire{
 		t: t,
 	}
 }
 
+// TODO (maybe):
+//  - ContainsHistoryEvents (should accept expectedHistory with and w/o event Ids)
+//  - StartsWithHistoryEvents (should accept expectedHistory with and w/o event Ids)
+//  - EndsWithHistoryEvents (should accept expectedHistory with and w/o event Ids)
+//  - WaitForHistoryEvents (call getHistory until expectedHistory is found, with interval and timeout)
+//  - Funcs like WithVersion, WithTime, WithAttributes, WithPayloadLimit(100) and pass them to PrintHistory
+//  - oneof support
+//  - enums as strings not as ints
+
 func (h HistoryRequire) EqualHistoryEvents(expectedHistory string, actualHistoryEvents []*historypb.HistoryEvent) {
 	if th, ok := h.t.(helper); ok {
 		th.Helper()
 	}
 
-	h.EqualHistory(expectedHistory, &historypb.History{Events: actualHistoryEvents})
+	expectedCompactHistory, expectedEventsAttributes := h.parseHistory(expectedHistory)
+	actualCompactHistory := h.formatHistoryEventsCompact(actualHistoryEvents)
+	require.Equal(h.t, expectedCompactHistory, actualCompactHistory)
+	for _, actualHistoryEvent := range actualHistoryEvents {
+		if expectedEventAttributes, ok := expectedEventsAttributes[actualHistoryEvent.EventId]; ok {
+			actualEventAttributes := reflect.ValueOf(actualHistoryEvent.Attributes).Elem().Field(0).Elem()
+			h.equalExpectedMapToActualAttributes(expectedEventAttributes, actualEventAttributes, actualHistoryEvent.EventId, "")
+		}
+	}
 }
 
 func (h HistoryRequire) EqualHistory(expectedHistory string, actualHistory *historypb.History) {
 	if th, ok := h.t.(helper); ok {
 		th.Helper()
 	}
 
-	expectedCompactHistory, expectedEventsAttributes := h.parseHistory(expectedHistory)
-	actualCompactHistory := h.formatHistoryCompact(actualHistory)
-	require.Equal(h.t, expectedCompactHistory, actualCompactHistory)
-	for _, actualHistoryEvent := range actualHistory.Events {
-		if expectedEventAttributes, ok := expectedEventsAttributes[actualHistoryEvent.EventId]; ok {
-			actualEventAttributes := reflect.ValueOf(actualHistoryEvent.Attributes).Elem().Field(0).Elem()
-			h.equalStructToMap(expectedEventAttributes, actualEventAttributes, actualHistoryEvent.EventId, "")
-		}
-	}
+	h.EqualHistoryEvents(expectedHistory, actualHistory.GetEvents())
 }
 
 func (h HistoryRequire) PrintHistory(history *historypb.History) {
-	_, _ = fmt.Println(h.formatHistory(history))
+	h.PrintHistoryEvents(history.GetEvents())
 }
 
 func (h HistoryRequire) PrintHistoryEvents(events []*historypb.HistoryEvent) {
-	h.PrintHistory(&historypb.History{Events: events})
+	_, _ = fmt.Println(h.formatHistoryEvents(events))
```

Comment:
If this is for testing then please use t.Logf so that it’s easily findable in the testing output